--- a/gobject/tests/Makefile.am	2012-03-19 01:02:40.826305315 +0400
+++ b/gobject/tests/Makefile.am	2012-03-19 01:04:33.148042013 +0400
@@ -21,10 +21,10 @@
 signals_SOURCES = signals.c marshalers.c
 
 marshalers.h: Makefile.am marshalers.list
-	$(AM_V_GEN) ../glib-genmarshal --prefix=test $(srcdir)/marshalers.list --header --valist-marshallers > marshalers.h
+	$(AM_V_GEN) $(GLIB_GENMARSHAL) --prefix=test $(srcdir)/marshalers.list --header --valist-marshallers > marshalers.h
 
 marshalers.c: Makefile.am marshalers.list
-	$(AM_V_GEN) ../glib-genmarshal --prefix=test $(srcdir)/marshalers.list --body --valist-marshallers > marshalers.c
+	$(AM_V_GEN) $(GLIB_GENMARSHAL) --prefix=test $(srcdir)/marshalers.list --body --valist-marshallers > marshalers.c
 
 BUILT_SOURCES = marshalers.h marshalers.c
 CLEANFILES = marshalers.h marshalers.c
