--- a/configure.in	2012-11-18 09:10:17.000000000 +0400
+++ b/configure.in	2012-11-18 09:15:38.000000000 +0400
@@ -1622,6 +1622,9 @@
   AC_MSG_NOTICE($external_ffmpeg_disabled)
   USE_EXTERNAL_FFMPEG=0
   AC_DEFINE([PIX_FMT_VDPAU_MPEG4_IN_AVUTIL], [1], [Whether AVUtil defines PIX_FMT_VDPAU_MPEG4.])
+
+  # check for yasm
+  AC_CHECK_PROG(HAVE_YASM,yasm,"yes","no",)
 fi
 
 echo "Checking for SWIG installation"
@@ -2599,6 +2602,7 @@
       `if test "$use_cpu" != "no"; then echo --cpu=$use_cpu; fi`\
       `if test "$use_neon" = "yes"; then echo --enable-neon; else echo --disable-neon; fi`\
       --target-os=$ffmpeg_target_os \
+      `if test "$HAVE_YASM" = "yes"; then echo --enable-yasm; fi` \
       --disable-muxers \
       --enable-muxer=spdif \
       --enable-muxer=adts \
