--- xbmc-orig/configure.in	2012-03-25 13:26:08.482499126 +0400
+++ xbmc-g45c575b/configure.in	2012-03-25 13:27:28.743740485 +0400
@@ -22,7 +22,7 @@ AC_DEFUN([XB_FIND_SONAME],
     AC_MSG_CHECKING([for lib$2 soname])
     $1_FILENAME=$($CC -nostdlib -o /dev/null $LDFLAGS -l$2 -Wl,-M 2>/dev/null | grep "^LOAD.*$2" | awk '{V=2; print $V}')
     if [[ ! -z $$1_FILENAME ]]; then
-      $1_SONAME=$(objdump -p $$1_FILENAME | grep "SONAME.*$2" | awk '{V=2; print $V}')
+      $1_SONAME=$($OBJDUMP -p $$1_FILENAME | grep "SONAME.*$2" | awk '{V=2; print $V}')
     fi
   else
     AC_MSG_CHECKING([for lib$2 dylib])
